JsonSchema/JsonSchema.cs

```diff
@@ -709,8 +852,27 @@ public override JsonSchema Read(ref Utf8JsonReader reader, Type typeToConvert, J
 						implementation = SchemaKeywordRegistry.GetNullValuedKeyword(keywordType) ??
 										 throw new InvalidOperationException($"No null instance registered for keyword `{keyword}`");
 					else
-						implementation = options.Read(ref reader, keywordType) as IJsonSchemaKeyword ??
-										 throw new InvalidOperationException($"Could not deserialize expected keyword `{keyword}`");
+					{
+#if NET6_0_OR_GREATER // JsonSerializer.Deserialize that takes a Utf8Reader isn't in the STJ that NetStandard 2.0 compiles against.
+						if (SchemaKeywordRegistry.TryGetTypeInfo(keywordType, out JsonTypeInfo? typeinfo))
+						{
+							implementation = JsonSerializer.Deserialize(ref reader, typeinfo!) as IJsonSchemaKeyword ??
+								throw new InvalidOperationException($"Could not deserialize expected keyword `{keyword}`");
+						}
+						else
+#endif
+						if (SchemaKeywordRegistry.RequiresDynamicSerialization)
+						{
+#pragma warning disable IL2026, IL3050 // Suppress because to get here the caller of SchemaKeywordRegistry must have suppressed the AOT warnings themselves
+							implementation = JsonSerializer.Deserialize(ref reader, keywordType) as IJsonSchemaKeyword ??
```

Comment:
Wouldn't we still want the `options.Read()` here?

Reply:
Yeah, I've gone back to the options.Read pattern which loses the static analysis but makes sure that this behaves the same. With the amount of test coverage that's here, I think we'll be ok.